src/compiler/program.ts

```diff
@@ -989,22 +989,60 @@ namespace ts {
 
         return program;
 
-        function resolveModuleNamesWorker(moduleNames: string[], containingFile: string, reusedNames?: string[], redirectedReference?: ResolvedProjectReference) {
+        function resolveModuleNamesWorker(moduleNames: string[], containingFile: SourceFile, reusedNames: string[] | undefined) {
+            if (!moduleNames.length) return [];

```

Comment:
This ensures that we aren't calculating redirect reference for files that contain no modules.